tests/integration/use-cache.test.ts

```diff
@@ -0,0 +1,429 @@
+    {
+      getDataTimeShouldShouldBeEqual,
+      resultWrapperComponentTimeShouldBeEqual,
+      pageComponentTimeShouldBeEqual,
+    }: ExpectedCachingBehaviorDefinition,
+  ) {
+    const $response1 = load(response1.body)
+    const $response2 = load(response2.body)
+
+    const getDataComparison = compareDates(
+      $response1,
+      $response2,
+      'getData-time',
+      getDataTimeShouldShouldBeEqual,
+      this.utils.diff,
+    )
+    const resultComponentComparison = compareDates(
+      $response1,
+      $response2,
+      'ResultWrapperComponent-time',
+      resultWrapperComponentTimeShouldBeEqual,
+      this.utils.diff,
+    )
+    const pageComponentComparison = compareDates(
+      $response1,
+      $response2,
+      'PageComponent-time',
+      pageComponentTimeShouldBeEqual,
+      this.utils.diff,
+    )
+
+    return {
+      pass:
+        getDataComparison.isExpected &&
+        resultComponentComparison.isExpected &&
+        pageComponentComparison.isExpected,
+      message: () =>
+        [getDataComparison.msg, resultComponentComparison.msg, pageComponentComparison.msg]
+          .filter(Boolean)
+          .join('\n\n'),
+    }
+  },
+})
+
+interface CustomMatchers<R = unknown> {
+  toBeCacheableResponse(): R
+  toHaveResponseCacheTag(tag: string): R
+  toHaveExpectedCachingBehavior(
+    response2: Awaited<InvokeFunctionResult>,
+    expectations: ExpectedCachingBehaviorDefinition,
+  ): R
+}
+
+declare module 'vitest' {
+  interface Assertion<T = any> extends CustomMatchers<T> {}
+}
+
+// Disable the verbose logging of the lambda-local runtime
+getLogger().level = 'alert'
+
+let ctx: FixtureTestContext
+beforeAll(async () => {
+  ctx = {
+    deployID: generateRandomObjectID(),
+    siteID: v4(),
+  } as FixtureTestContext
+
+  vi.stubEnv('SITE_ID', ctx.siteID)
+  vi.stubEnv('DEPLOY_ID', ctx.deployID)
+  vi.stubEnv('NETLIFY_PURGE_API_TOKEN', 'fake-token')
+  await startMockBlobStore(ctx as FixtureTestContext)
+
+  await createFixture('use-cache', ctx)
+  await runPlugin(ctx)
+})
+
+// only supporting latest variant (https://github.com/vercel/next.js/pull/76687)
+// first released in v15.3.0-canary.13 so we should not run tests on older next versions
+describe.skipIf(!nextVersionSatisfies('>=15.3.0-canary.13'))('use cache', () => {
+  describe('default (in-memory cache entries, shared tag manifests)', () => {
+    for (const {
+      expectedCachingBehaviorWhenUseCacheRegenerates,
+      useCacheLocationLabel,
+      useCacheLocationPathSegment,
+      useCacheTagPrefix,
+    } of [
+      {
+        useCacheLocationLabel: "'use cache' in data fetching function",
+        useCacheLocationPathSegment: 'use-cache-data',
+        useCacheTagPrefix: 'data',
+        expectedCachingBehaviorWhenUseCacheRegenerates: {
+          // getData function has 'use cache' so it should report same generation time, everything else is dynamically regenerated on each request
+          getDataTimeShouldShouldBeEqual: true,
```

Comment:
Don't believe the double `Should` is intentional

Reply:
fixed typo in https://github.com/opennextjs/opennextjs-netlify/pull/2862/commits/e18b9b8766ce3b73b3e824198308dc0e39dde7cf 🤞